File: 09 - Py-Js/ques01.py
import math
import sys

a = int(input("Enter a: "))
b = int(input("Enter b: "))
c = int(input("Enter c: "))
eqn = f"{a}x^2 + {b}x + {c} = 0"
print(eqn)

d = b**2 - 4 * a * c

if d < 0:
    print("no roots exist")
    sys.exit()

elif d > 0:
    r1 = (-b + math.sqrt(d)) / 2 * a
    r2 = (-b - math.sqrt(d)) / 2 * a
    print("root1", r1)
    print("root2", r2)
else:
    r = (-b + math.sqrt(d)) * 2 * a
    print("root", r)

File: 09 - Py-Js/ques01.js
function quadratic_roots(a, b, c) {
    console.log("eqn", `${a}x^2 + ${b}x + ${c}`)
    let d = b ** 2 - 4 * a * c
    let r1 = (-b + Math.sqrt(d)) / 2 * a
    let r2 = (-b - Math.sqrt(d)) / 2 * a

    if (d < 0) return
    else if (d > 0) {
        console.log(`root1 = ${r1} and root2 = ${r2}`)
    }
    else console.log(`repeated root: ${r1}`)
}

quadratic_roots(1, -5, 6)
quadratic_roots(1, -4, 4)